### PR TITLE
#121: modify ordering strategies to remove use of load average

### LINKED ISSUE #121: Study ordering strategies integrating communication weights
The goal of this issue is to study and discuss improved ordering strategies based on total work, to replace line 3 below:

Screen Shot 2021-12-09 at 11 39 00 AM

### PR DESCRIPTION
**Pull request by ppebay**
Fixes #121 (cf. same issue for discussion)

### CODE REVIEW

**Comment by ppebay** [comment]
With `increasing_times` ordering (new):
Screen Shot 2022-02-13 at 6 28 45 PM

**Comment by ppebay** [comment]
This PR also implements the computation and reporting (in debug logging mode) of arrangements to allow for direct comparison with `rank-object-enumerator` outputs:
```
python LBAF.py |& grep arrangement
[lbsRuntime] Iteration 0 arrangement: (0, 0, 0, 0, 1, 1, 1, 1, 2)
[lbsRuntime] Iteration 1 arrangement: (0, 3, 3, 2, 3, 1, 0, 3, 2)
[lbsRuntime] Iteration 2 arrangement: (0, 3, 3, 2, 3, 1, 0, 3, 2)
[lbsRuntime] Iteration 3 arrangement: (0, 3, 3, 2, 3, 1, 0, 3, 2)
[lbsRuntime] Iteration 4 arrangement: (0, 3, 3, 2, 3, 1, 0, 3, 2)
[lbsRuntime] Iteration 5 arrangement: (0, 3, 3, 2, 3, 1, 0, 3, 2)
[lbsRuntime] Iteration 6 arrangement: (0, 3, 3, 2, 3, 1, 0, 3, 2)
[lbsRuntime] Iteration 7 arrangement: (0, 3, 3, 2, 3, 1, 0, 3, 2)
[lbsRuntime] Iteration 8 arrangement: (0, 3, 3, 2, 3, 1, 0, 3, 2)
```

**Review by nlslatt — COMMENTED** [review]
Should we really be throwing out all the old functionality? It seems like we should have made allowing multiple algorithms a higher priority before making huge changes.

File: src/Applications/conf.yaml [hunk 1]
```diff
@@ -9,6 +9,10 @@
 #    parameters: optional parameters that are specitic to each criterion
 # deterministic_transfer    decide whether transfer must be deterministic
 #    type: Boolean (default: False)
+# order_strategy            description of ordering of objects for transfer
+#    type: str in arbitrary (default), element_id, increasing_times, decreasing_times, best_migrations
```
[the remark this answers is not in the document]
Comment (ppebay) [reply src/Applications/conf.yaml]: Thank you, fixed!

**Comment by ppebay** [comment]
Thanks @nlslatt for the comment; my initial thinking was that we would try to re-design `fewest_migrations` and `small_objects` in the new context, and re-introduce them later. But I agree with your remark and I have re-introduced them (with the slightly modified API taking an extra parameter), "as-is", with a reference to the average load. So effectively they are available as before but are not adapted to the generalized notion of "work" and we can re-work them later (no pun intended). Creating an issue for that. Thank you.

**Pull request merged**